### PR TITLE
Fix: Update policy mapping properly in notif'n

### PR DESCRIPTION
**Pull request by donatello**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

This is fixing a regression from an earlier change where STS account loading was made lazy.

## Motivation and Context

Fix a bug in AssumeRoleWithCertificate where created credentials dont work with other nodes than the one in which it was created.

## How to test this PR?

1. Setup multi-node MinIO with a self signed certificate (I used https://github.com/FiloSottile/mkcert) and enable AssumeRoleWithCertificate - https://github.com/minio/minio/blob/master/docs/sts/tls.md
2. Create client cert (some openssl commands for this are here https://docs.aws.amazon.com/iot/latest/developerguide/create-device-cert.html - use the self signed CA used in the previous step).
3. Generate creds with AssumeRoleWithCert - example curl command here - https://github.com/minio/minio/blob/master/docs/sts/tls.md#example
4. Test creds against each node of minio - it will work only with the node on which it was created - and after apply this fixes, repeat the test and it will work on all nodes.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression (https://github.com/minio/minio/pull/17994)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/iam-store.go [hunk 1]
```diff
@@ -1626,37 +1626,38 @@ func (store *IAMStoreSys) UserNotificationHandler(ctx context.Context, accessKey
...
-		u, ok := cache.iamSTSAccountsMap[accessKey]
-		if ok {
-			cred := u.Credentials
-			if cred.ParentUser != "" && cred.ParentUser != globalActiveCred.AccessKey {
-				if _, ok := cache.iamUserPolicyMap[cred.ParentUser]; !ok {
-					cache.iamUserPolicyMap[cred.ParentUser] = cache.iamSTSPolicyMap[accessKey]
-					cache.updatedAt = time.Now()
-				}
-			}
-		}
```
Comment (donatello) [inline cmd/iam-store.go:1659]: I don't think we need this block any more - not sure what the original purpose was.

Comment (harshavardhana) [reply cmd/iam-store.go:1659]: This block for OIDC credentials

Comment (donatello) [reply cmd/iam-store.go:1659]: Yeah, I don't think we need it - for sts accounts we lookup iamSTSPolicyMap for the parent (which is "virtual").

File: cmd/iam-store.go [hunk 2]
```diff
@@ -1583,6 +1583,10 @@ func (store *IAMStoreSys) UserNotificationHandler(ctx context.Context, accessKey
 		m = cache.iamUsersMap
 	}
 	err := store.loadUser(ctx, accessKey, userType, m)
+	defer func() {
+		cache.updatedAt = time.Now()
```
Comment (vadmeste) [inline cmd/iam-store.go]: It makes sense for me to only update cache.updateAt when this call returns a nil error, /cc @poornas

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/iam-store.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/iam-store.go]: yeah that makes sense @donatello ^^

Comment (donatello) [reply cmd/iam-store.go]: That's not completely correct - we do in some cases update the cache and still return non-nil error. I am slightly updating this logic.

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**